services/director-v2/src/simcore_service_director_v2/core/settings.py

```diff
@@ -159,9 +160,13 @@ def endpoint(self) -> str:
 
 class DynamicSidecarProxySettings(BaseCustomSettings):
     DYNAMIC_SIDECAR_CADDY_VERSION: str = Field(
-        "2.4.5-alpine",
+        "2.6.4-alpine",
         description="current version of the Caddy image to be pulled and used from dockerhub",
     )
+    DYNAMIC_SIDECAR_CADDY_ADMIN_API_PORT: PortInt = Field(
+        default_factory=lambda: random.randint(1025, 65535),
```

Comment:
Why a random port, that is a very good question. 
Keep in mind the following:
- the proxy is started without a configuration, just the admin API listening on 0.0.0.0:RANDOM_PORT
- it awaits configuration, which happens after the user services are started.
- once the configuration is sent, the admin interface is no longer available (it is disabled) the proxy configuration can no  longer be changed.

There is a small interval from where the user services are started and they are in `running` state, that the admin interface is stil available. After the user services are in running, the proxy is reconfigured (admin interface goes away). Afterwards the interface is displayed to the user.
During this period some code present int the boot portion of the service could run that can interfere with the proxy's admin interface. To make this harder the port is chose randomly.

Please note: there is no way to protect the admin route of caddy, I have already tried and failed 3 times.

---

Regarding the "free port" question:
- this port is picked by the director-v2 or at most it will be picked by the dy-sidecar in the future; not by service which will be running it. So it can't be done
- also the port is picked randomly in the interval of freely usable ports which are not used by any other system services for that container.